dolphin/drivers/api.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from oslo_log import log
+from oslo_utils import uuidutils
+
+from dolphin.drivers import manager
+
+LOG = log.getLogger(__name__)
+
+
+class API(object):
+    def __init__(self):
+        self.driver_manager = manager.DriverManager()
+
+    @staticmethod
+    def get_storage_registry():
+        """Show register parameters which the driver needs."""
+        pass
+
+    def register_storage(self, context, access_info):
+        """Discovery a storage system with access information."""
+        access_info['storage_id'] = six.text_type(uuidutils.generate_uuid())
```

Comment:
What will happen if the caller sends access_inf['storage_id']? It will always be overwritten, right?

Reply:
Currently, the upper level did not create storage_id, so it's created here.

Reply:
ok